core/src/main/java/org/fao/geonet/resources/CMISUtils.java

```diff
@@ -241,7 +253,7 @@ public Document saveDocument(String key, CmisObject cmisObject, Map<String, Obje
 
                 if (cmisConfiguration.existSecondaryProperty()) {
                     //need to reload document to avoid  "Document is not the latest version" when updating secondary types.
-                    doc = (Document) cmisConfiguration.getClient().getObjectByPath(key, oc);
+                    doc.refresh();
```

Comment:
I understand that `doc` already has a value and the old code was intended to reload it, but it's more appropriate to use `refresh`, correct?

Reply:
Yes you are correct. By using refresh() it can issue a get document by id.  Otherwise it was getting the document by path which is slower.